core/drivers/crypto/caam/caam_rng.c

```diff
@@ -541,17 +539,39 @@ enum caam_status caam_rng_init(vaddr_t ctrl_addr)
 		retstatus = caam_rng_instantiation();
 	}
 
-#ifdef CFG_NXP_CAAM_RNG_DRV
 	if (retstatus == CAAM_NO_ERROR)
 		retstatus = caam_rng_init_data();
-#endif
 
 	if (retstatus != CAAM_NO_ERROR)
 		do_free();
 
 	return retstatus;
 }
 
+#define RNG_SEED_LEN	64
+
+#ifndef CFG_NXP_CAAM_RNG_DRV
+void plat_rng_init(void)
+{
+	TEE_Result res = TEE_SUCCESS;
+	uint8_t buf[RNG_SEED_LEN] = { };
+
+	res = do_rng_read((uint8_t *)&buf, RNG_SEED_LEN);
```

Comment:
`	res = do_rng_read(buf, sizeof(buf));` should be fine (`sizeof()` for consistency with below).